main/observing/observation_run.py

```diff
@@ -630,3 +652,70 @@ def _shutdown_procedure(self, calibration, cooler=True):
             self.take_calibration_images()
         if cooler:
             self.camera.onThread(self.camera.cooler_set, False)
+        logging.info('Stopping thread monitoring')
+        self.monitor.run_th_monitor = False
+        logging.debug('Number of thread restarts: {}'.format(self.monitor.n_restarts))
```

Comment:
I saw you added this to the stop_threads method but you still need to get rid of this